### ISSUE TITLE
Some issues on summarization example

### ISSUE THREAD

**Issue by Shentao-YANG** [huggingface/transformers#20972]
@patil-suraj 
Thanks for the beautiful example in `main/examples/pytorch/summarization`. I do, however, have the following issues with the `run_summarization_no_trainer.py` file.
1. The flag `--max_length` seems not used. 
2. The checking `check_min_version("4.26.0.dev0")` seems ahead of the current release of `4.25.1`
3. This file does not have the flags no `test_file`, `max_train/eval/predict_samples`, while `run_summarization.py` has.
4. Also it will be helpful to add the functionality of gradient norm clipping.

Thanks for your help.
